src/4_calculator/app.js

```diff
@@ -0,0 +1,152 @@
+const $window = document.querySelector("#window");
+const $buttons = document.querySelector("#button-container");
+
+let num1 = "";
+let num2 = "";
+let operator = "";
+let result = "";
+
+function handleClickButton(event) {
+  const className = event.target.className;
+  const textContent = event.target.textContent.trim();
+
+  if (className.includes("clear")) {
+    handleClickAllClear();
+    return;
+  }
+
+  if (className.includes("number")) {
+    handleClickNumber(textContent);
+    return;
+  }
+
+  if (className.includes("operator")) {
+    handleClickOperator(textContent);
+    return;
+  }
+
+  if (className.includes("back-space")) {
+    handleClickBackSpace(textContent);
+    return;
+  }
+}
+
+function handleClickAllClear() {
+  $window.textContent = "0";
+  [num1, num2, operator, result] = ["", "", "", ""];
```

Comment:
이 코드는 모든 값을 초기화 시키는 의미로 해석됩니다. 여기서 제가 갖는 몇가지 의문점이 있습니다.

1.  가독성
저 코드를 보고 초기화시키는 코드라고 해석할 수 있습니다. 그런데 같이 드는 생각은 굳이 배열 안에 저 변수를 집어넣어서 초기화를 시켜야하는 이유가 뭔지 궁금해 집니다.
아래코드가 없어보이지만(?) 직관적일 수 있습니다.
```suggestion
num1 = "", num2 = "", operator = "", result = "";
```

2. 메모리 관점
이 부분은 스터디 시간에  설명하는 편이 좀 더 나을거 같습니다 :)